code/modules/mob/living/simple_animal/bot/secbot.dm

```diff
@@ -449,6 +449,24 @@
 	new /obj/effect/decal/cleanable/blood/oil(loc)
 	..()
 
+///Disassembling the bot in a civilized manner with a multitool
+/mob/living/simple_animal/bot/secbot/disassemble()
+	walk_to(src,0)
+	var/turf/Tsec = get_turf(src)
+	var/obj/item/secbot_assembly/Sa = new /obj/item/secbot_assembly(Tsec)
+	Sa.build_step = 1
+	Sa.overlays += "hs_hole"
+	Sa.created_name = name
+	new /obj/item/assembly/prox_sensor(Tsec)
+	new /obj/item/melee/baton(Tsec)
+	drop_part(robot_arm, Tsec)
+	if((src) == /mob/living/simple_animal/bot/secbot/beepsky/)
```

Comment:
```suggestion
	if(istype(src, /mob/living/simple_animal/bot/secbot/beepsky)
```
Comparing an object to a typepath doesn't work, you have to use `istype()`!
Also, don't forget to test as much of your PR as you can!

Reply:
Ah I had forgotten to test if beepsky actually dropped it's things after writing it. Thank you!